core/directory.go

```diff
@@ -496,6 +498,68 @@ func (dir *Directory) WithNewFileDagOp(ctx context.Context, dest string, content
 	return dir, nil
 }
 
+func (dir *Directory) WithPatch(ctx context.Context, patch string) (*Directory, error) {
+	dir = dir.Clone()
+
+	parentRef, err := getRefOrEvaluate(ctx, dir.Result, dir)
+	if err != nil {
+		return nil, err
+	}
+
+	bkSessionGroup, ok := buildkit.CurrentBuildkitSessionGroup(ctx)
+	if !ok {
+		return nil, fmt.Errorf("no buildkit session group in context")
+	}
+
+	opt, ok := buildkit.CurrentOpOpts(ctx)
+	if !ok {
+		return nil, fmt.Errorf("no buildkit opts in context")
+	}
+
+	query, err := CurrentQuery(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx = trace.ContextWithSpanContext(ctx, opt.CauseCtx)
+	stdio := telemetry.SpanStdio(ctx, InstrumentationLibrary)
+	defer stdio.Close()
+
+	newRef, err := query.BuildkitCache().New(ctx, parentRef, bkSessionGroup, bkcache.WithRecordType(bkclient.UsageRecordTypeRegular),
+		bkcache.WithDescription("patch"))
+	if err != nil {
+		return nil, err
+	}
+	err = MountRef(ctx, newRef, bkSessionGroup, func(root string) (rerr error) {
+		var stdout, stderr strings.Builder
+		apply := exec.Command("git", "apply", "-")
```

Comment:
Just double-checking, this executes the `git` binary on the host? Is this a new runtime dependency, or did it already exist?

Not a dealbreaker, just curious as it has implications for packaging.